python/sglang/srt/layers/quantization/int4fp8_moe.py

```diff
@@ -0,0 +1,618 @@
+    from aiter.fused_moe_bf16_asm import ck_moe_2stages
+    from aiter.ops.shuffle import shuffle_weight
+
+    ON_GFX950 = "gfx950" in torch.cuda.get_device_properties("cuda").gcnArchName
+
+logger = logging.getLogger(__name__)
+
+# explicitly use pure text format, with a newline at the end
+# this makes it impossible to see the animation in the progress bar
+# but will avoid messing up with ray or multiprocessing, which wraps
+# each line of output with some prefix.
+_BAR_FORMAT = "{desc}: {percentage:3.0f}% Completed | {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_fmt}]\n"  # noqa: E501
+
+def tqdm_reset_no_print(tqdm_bar: tqdm, total=None):
+    tqdm_bar.n = 0
+    if total is not None:
+        tqdm_bar.total = total
+    if tqdm_bar.disable:
+        return
+    tqdm_bar.last_print_n = 0
+    tqdm_bar.last_print_t = tqdm_bar.start_t = tqdm_bar._time()
+    tqdm_bar._ema_dn = EMA(tqdm_bar.smoothing)
+    tqdm_bar._ema_dt = EMA(tqdm_bar.smoothing)
+    tqdm_bar._ema_miniters = EMA(tqdm_bar.smoothing)
+
+class QuarkInt4Fp8Config(QuantizationConfig):
+    """Config class for Quark Quantization.
+
+    - Weight: static, per-channel, symmetric
+    - Activation: dynamic, per-token, symmetric
+    """
+
+    def __init__(self):
+        self.activation_scheme = "dynamic"
+        self.weight_block_size = None
+
+        self.num_quant_layers = 0
+
+        tp_rank = get_tensor_model_parallel_rank()
+
+        # The weight iterator already has a progress bar on rank=0, account for that.
+        position = 1 + tqdm._get_free_pos()
+        self.online_quant_progress_bar = tqdm(total=0, desc=f"Online int4fp8_moe quantization on rank={tp_rank}", position=position, bar_format=_BAR_FORMAT, mininterval=2.)
+
+    @classmethod
+    def get_supported_act_dtypes(cls) -> List[torch.dtype]:
+        return [torch.float16, torch.bfloat16]
+
+    @classmethod
+    def get_min_capability(cls) -> int:
+        return 70
+
+    @classmethod
+    def get_name(self) -> str:
+        return "int4fp8_moe"
+
+    @classmethod
+    def get_config_filenames(cls) -> List[str]:
+        return []
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "QuarkInt4Fp8Config":
+        return cls()
+
+    def get_quant_method(
+        self,
+        layer: torch.nn.Module,
+        prefix: str,
+    ) -> Optional["QuantizeMethodBase"]:
+        # TODO: fix circular imports issues in sglang forcing us to import here instead of at
+        # the top of file.
+        from sglang.srt.layers.linear import LinearBase
+        from sglang.srt.layers.moe.fused_moe_triton import FusedMoE
+
+        if isinstance(layer, LinearBase):
+            return QuarkInt4Fp8LinearMethod(self)
+        elif isinstance(layer, FusedMoE):
+            return QuarkInt4Fp8MoEMethod(self)
+        
+        return None
+
+    def get_scaled_act_names(self) -> List[str]:
+        return []
+
+
+class QuarkInt4Fp8LinearMethod(LinearMethodBase):
```

Comment:
IIUC the `Int4Fp8` in name only refers to the spec name, For linear layers this just performs fp8 quantization and no int4 right? Let's put a note in the docstring if that's the case.